core/conversion/converters/impl/select.cpp

```diff
@@ -202,6 +230,65 @@ auto select_registrations TORCHTRT_UNUSED =
 
                return true;
              }})
+        .pattern({"aten::roll(Tensor self, int[1] shifts, int[1] dims=[]) -> (Tensor)",
+                  [](ConversionCtx* ctx, const torch::jit::Node* n, args& args) -> bool {
+                    auto in = args[0].ITensor();
+                    auto shifts = args[1].unwrapToIntList().vec();
+                    auto dims = args[2].unwrapToIntList().vec();
+
+                    TORCHTRT_CHECK(dims.size() == shifts.size(), "dims.size() should be equal to shifts.size()");
+                    if (ctx->input_is_dynamic) {
+                      TORCHTRT_THROW_ERROR("aten::roll is currently not support in dynamic input shape compilation");
+                    } else {
+                      auto in_shape = util::toVec(in->getDimensions());
+                      for (size_t i = 0; i < dims.size(); i++) {
+                        auto dim = dims[i] < 0 ? (in_shape.size() + dims[i]) : dims[i];
+                        TORCHTRT_CHECK(dim < in_shape.size(), "Dimension out of range");
+                        in = roll(ctx, in, shifts[i], dim, in_shape);
+                      }
+                      auto out = ctx->AssociateValueAndTensor(n->outputs()[0], in);
+
+                      LOG_DEBUG("Output tensor shape: " << out->getDimensions());
+
+                      return true;
+                    }
+                  }})
+        .pattern({"aten::index.Tensor(Tensor self, Tensor?[] indices) -> (Tensor)",
+                  [](ConversionCtx* ctx, const torch::jit::Node* n, args& args) -> bool {
+                    auto in = args[0].ITensorOrFreeze(ctx);
+                    auto ts = args[1].IValue()->toListRef();
+
+                    std::vector<nvinfer1::ITensor*> tensors;
+                    for (auto t : ts) {
+                      if (t.isTensor()) {
+                        auto torch_tensor = t.toTensor();
+                        tensors.push_back(tensor_to_const(ctx, torch_tensor));
+                      } else {
+                        auto cont = t.toCustomClass<TensorContainer>();
+                        tensors.push_back(cont->tensor());
+                      }
+                    }
+
+                    TORCHTRT_CHECK(
+                        tensors.size() == 1,
+                        "This version of Torch-TensorRT only supports one index in aten::index.Tensor");
```

Comment:
Can you clarify what this means? Can I only do something like tensor[0] or something?

Reply:
In Pytorch, 'aten::index.Tensor' can receive several indices, like this:
```
x = torch.rand(5, 7, 10)
idx0 = torch.tensor([0, 0, 0, 0])
idx1 = torch.tensor([1, 1, 1, 1])
idx2 = torch.tensor([2, 2, 2, 2])
y1 = x[idx0, idx1, idx2] # result is like [x[0, 1, 2], x[0, 1, 2], x[0, 1, 2], x[0, 1, 2]]
y2 = x[idx0] # result is like [x[0, :, :], x[0, :, :], x[0, :, :], x[0, :, :]]
```
I only support one index as input currently like `y2 = x[idx0]`

Reply:
> In Pytorch, 'aten::index.Tensor' can receive several indices, like this:
> 
> ```
> x = torch.rand(5, 7, 10)
> idx0 = torch.tensor([0, 0, 0, 0])
> idx1 = torch.tensor([1, 1, 1, 1])
> idx2 = torch.tensor([2, 2, 2, 2])
> y1 = x[idx0, idx1, idx2] # result is like tensor([x[0, 1, 2], x[0, 1, 2], x[0, 1, 2], x[0, 1, 2]])
> y2 = x[idx0] # result is like tensor([x[0, :, :], x[0, :, :], x[0, :, :], x[0, :, :]])
> ```
> 
> I only support one index as input currently like `y2 = x[idx0]`. We can only process tensors[0].

Hi @narendasan Is this explanation clear?

Reply:
Ok can we improve this message to convey what this means

Reply:
Hi @narendasan i have added some annotation and improved this message.